src/main/java/net/apnic/rdap/iana/scraper/IANABootstrapScraper.java

```diff
@@ -0,0 +1,320 @@
+        }
+        catch(URISyntaxException ex)
+        {
+            LOGGER.log(Level.SEVERE, "Exception when generating IANA url's",
+                       ex);
+            throw new RuntimeException(ex);
+        }
+        finally
+        {
+            ASN_URI = asnURI;
+            DOMAIN_URI = domainURI;
+            IPV4_URI = ipv4URI;
+            IPV6_URI = ipv6URI;
+        }
+    }
+
+    /**
+     * Constructor for creating an IANA bootstrap scraper.
+     *
+     * @param authorityStore Store to find authorities
+     * @param asnStore Store for emplacing asn resource mappings
+     * @param domainStore Store for emplacing domain resource mappings
+     * @param ipStore Store for emplacing ip resource mappings
+     */
+    public IANABootstrapScraper(RDAPAuthorityStore authorityStore,
+                                ResourceStore<AsnRange> asnStore,
+                                ResourceStore<Domain> domainStore,
+                                ResourceStore<IpRange> ipStore)
+    {
+        this.authorityStore = authorityStore;
+        this.asnStore = asnStore;
+        this.domainStore = domainStore;
+        this.ipStore = ipStore;
+        restClient = new AsyncRestTemplate();
+        setupRequestHeaders();
+    }
+
+    /**
+     * {@inheritDocs}
+     */
+    @Override
+    public String getName()
+    {
+        return "iana-bootstrap-scraper";
+    }
+
+    /**
+     * Sets up common HTTP headers used in every request to the IANA bootstrap
+     * service.
+     */
+    private void setupRequestHeaders()
+    {
+        requestHeaders = new HttpHeaders();
+        requestHeaders.setAccept(Arrays.asList(MediaType.APPLICATION_JSON));
+        requestHeaders.add(HttpHeaders.USER_AGENT, "");
+    }
+
+    /**
+     * Main scraper method that quicks off a one time scrape of IANA.
+     *
+     * This method is trigger by a scraper scheduler.
+     *
+     * @see net.apnic.rdap.scraper.ScraperScheduler
+     */
+    @Override
+    public CompletableFuture<Void> start()
+    {
+        return CompletableFuture.allOf(updateASNData(), updateDomainData(),
+                                       updateIPv4Data(), updateIPv6Data());
+    }
+
+    /**
+     * Grunt work method that performs all bootstrap http requests and returns
+     * the servers http response.
+     *
+     * Assumes per the RFC that all requests are going to be GET.
+     *
+     * @param bootStrapURI The URI to call as apart of this request.
+     * @return Promise of the server response with a JSON body.
+     */
+    private CompletableFuture<ResponseEntity<JsonNode>>
+        makeBootstrapRequest(URI bootStrapURI)
+    {
+        HttpEntity<?> entity = new HttpEntity<>(requestHeaders);
+
+        ListenableFuture<ResponseEntity<JsonNode>> lFuture =
+            restClient.exchange(bootStrapURI, HttpMethod.GET,
+                                entity, JsonNode.class);
+        return ConcurrentUtil.buildCompletableFuture(lFuture);
```

Comment:
(Suggestion) Rather than adapt spring's ListenableFuture into a CompletableFuture, you may as well just use a regular restTemplate and call CompletableFuture.supplyAsync(). Id also recommend injecting an Executor, since this can be useful - such as running with a single thread for unit testing (note that AsycRestTemplate will build itself a AsyncListenableTaskExecutor if you don't pass it one).